### PR TITLE
[ML-24419] Updated artifact downloads to pass in http headers

### PR DESCRIPTION
**Pull request by kriscon-db**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

We add the passthrough of encryption headers recieved from MLflow service into the download request for both artifacts and models.

## How is this patch tested?

I modified existing tests to verify that the headers are passed through properly.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/artifact/databricks_models_artifact_repo.py [hunk 1]
```diff
@@ -110,12 +110,19 @@ def _get_signed_download_uri(self, path=None):
...
                 " status code %s. Response body: %s" % (path, response.status_code, response.text)
             )
-        return json_response.get("signed_uri", None)
+        return json_response.get("signed_uri", None), json_response.get("headers", None)
+
+    @staticmethod
+    def _process_raw_headers(raw_headers):
+        if raw_headers == None or len(raw_headers) == 0:
+            return {}
+        return {cur_header.get("key"): cur_header.get("value") for cur_header in raw_headers}
```
Comment (kriscon-db) [inline mlflow/store/artifact/databricks_models_artifact_repo.py:119]: not sure how defensive we want to be here about malformed header responses/changes to the header structure.

Comment (dbczumar) [reply mlflow/store/artifact/databricks_models_artifact_repo.py:119]: `DatabricksArtifactRepo._convert_http_headers` seems more robust. Can we use that implementation? For example, `cur_header.get("key")` will return `None` if `key` isn't defined as opposed to skipping the entry.

**Review by dbczumar — APPROVED** [review]
LGTM! Can we merge this into a feature branch? Happy to stamp a separate PR :)

File: mlflow/store/artifact/databricks_models_artifact_repo.py (hunk 1 above)
Comment (kriscon-db) [reply mlflow/store/artifact/databricks_models_artifact_repo.py:119]: I can do this but I'll need to patch the universe protobuf definition of HttpHeader to match the one that the databricks artifact repo is using.  I think there is a slight difference in field naming.

Comment (dbczumar) [reply mlflow/store/artifact/databricks_models_artifact_repo.py:120]: I see - no need in that case. Separate method is fine, but let's make sure it's similarly robust to issues

Comment (kriscon-db) [reply mlflow/store/artifact/databricks_models_artifact_repo.py]: I have a PR to fix the headers in universe.  I had originally made that method in DatabricksArtifactRepo static but I don't actually like how that looks and have now backed out and we just have convenience methods in both classes.  If we really want, we can set up a utils.py file and have both classes read from it, but we can do that in a follow on commit.

**Pull request merged**